src/v/cloud_storage/tests/remote_partition_test.cc

```diff
@@ -198,32 +358,43 @@ static void print_segments(const std::vector<in_memory_segment>& segments) {
     }
 }
 
+enum class manifest_inconsistency {
```

Comment:
should we test with a gap between segments too? (possibly with some non-data batches in the gap too)

Reply:
We actually test this with gaps but we don't have non-data batches in that test. We also need to add non-data batches that doesn't affect offset translation to the mix (like tx_fence, tx_commit, etc).